FreeRTOS_TCP_IP.c

```diff
@@ -3623,13 +3623,19 @@
                     }
                     else
                     {
+                        ProtocolHeaders_t * pxProtocolHeaders;
```

Comment:
Just a thought: We might want to avoid mixed code and declaration. I know this makes code more readable - but we still might want to refrain from doing this.

Reply:
> Just a thought: We might want to avoid mixed code and declaration. I know this makes code more readable - but we still might want to refrain from doing this.

I agree that mixing code with declarations decreases the quality, and mistakes will become more likely.  But `pxProtocolHeaders` was declared at the beginning of a compound block.  There are many places in the +TCP code where local variables are declared at the beginning of a compound block, thus limiting their scope, which I think is good.

However, I didn't like the long declaration neither, and want to clear the TCP flags with an indexed assignment:
~~~
    ucLastPacket[ uxOffset + ipTCP_FLAGS_OFFSET ] = tcpTCP_FLAG_ACK;
~~~
where `ipTCP_FLAGS_OFFSET` is defined as 13U. This will also work for IPv6.

Reply:
PS. I just retested the solution, by running it twice: with and without clearing the TCP flags in `ucLastPacket`. It worked as expected.